addons/mail/models/discuss/mail_message.py

```diff
@@ -24,3 +24,13 @@ def _bus_channel(self):
         if self.env.user._is_public() and guest:
             return guest._bus_channel()
         return super()._bus_channel()
+
+    def _get_forbidden_access_extra(self, forbidden, operation, documents, model, messages_to_check):
+        # Handle discuss channel read only
+        if model == "discuss.channel" and operation in ["write", "create"]:
+            for document in documents:
+                # sudo: we need to check this status, even if the user
+                # cannot read the thread.
+                if document.sudo().read_only and not self.env.user._is_admin():
+                    forbidden += self.browse(messages_to_check)
+        return forbidden
```

Comment:
This doesn't really make use of `self` as message.

Maybe the method would be better on thread (and overridden in channel) which would remove the need to check/pass the model and documents params.